crates/neon-sys/src/lib.rs

```diff
@@ -18,8 +18,14 @@ pub struct Local {
 /// the function is bound to and a mechanism to return a value to the caller.
 pub type FunctionCallbackInfo = c_void;
 
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Isolate__ {
```

Comment:
This is the pattern that `bindgen` uses to give raw pointers specific times instead of `c_void`. It works really well for letting the borrow checker find the mismatches between `node_env` and `Isolate`.